beluga/test/beluga/random/test_multivariate_uniform_distribution.cpp

```diff
@@ -26,19 +26,48 @@
 namespace {
 
 using beluga::testing::Vector2Near;
+using beluga::testing::Vector3Near;
 
-TEST(UniformFreeSpaceGridDistribution, SingleSlot) {
+TEST(MultivariateUniformDistribution, BoundingRegion2d) {
+  constexpr double kTolerance = 0.01;
+  auto region = beluga::BoundingRegion2d{};
+  region.x_min = 3.00;
+  region.x_max = region.x_min + kTolerance;
+  region.y_min = -2.00;
+  region.y_max = region.y_min + kTolerance;
+  auto distribution = beluga::MultivariateUniformDistribution{region};
+  auto engine = std::mt19937{std::random_device()()};
```

Comment:
@nahueespinosa meta: should we fix the seed for tests to be reproducible? Same elsewhere.

Reply:
I don't think so. I found this article related to the usage of seeds in ML projects: https://misailo.cs.illinois.edu/papers/icst22.pdf

I found the recommendations useful:
- Use fixed seeds only for tests checking for exact reproducibility of some functionality in their code.
- Randomize and log the seeds in other tests for nondeterministic algorithms to allow both reproducibility of failures and diverse executions.
- Use Test Re-runs on failure instead of setting seeds to mitigate random failures.
- Determine optimal test settings such as hyper-parameters for the algorithm(s) under test and/or assertion bounds to minimize flakiness.

I think we should treat flakiness as a bug. Logging the used seed on failure is a nice feature, but it exceeds the scope of this PR.

Reply:
Interesting read. I'm satisfied with the reasoning.

> Logging the used seed on failure is a nice feature, but it exceeds the scope of this PR.

Fair enough. I suspect we'll add it when and if this comes back to haunt us.